File: netlify/functions/lib/scrapers/torrentquest.js
/**
 * TorrentQuest scraper - HTML scraping.
 */
const client = require('../client');
const { parseSeeders, parseSize, cleanName } = require('../utils');

const NAME = 'TorrentQuest';
const BASE = 'https://torrentquest.com';

async function search(query) {
  const url = `${BASE}/search?q=${encodeURIComponent(query)}&sort=seeders&order=desc`;
  const $ = await client.getSoup(url);
  if (!$) return [];

  const results = [];
  $('table tbody tr, div.search-result').each((_, row) => {
    const $row = $(row);
    const magnetTag = $row.find('a[href^="magnet:"]').first();
    if (!magnetTag.length) return;
    const magnet = magnetTag.attr('href');

    let name = '';
    const nameTag = $row.find('a.title, td a[href*="/torrent/"]').first();
    if (nameTag.length) name = cleanName(nameTag.text());
    if (!name) {
      const dnMatch = (magnet || '').match(/dn=([^&]+)/);
      if (dnMatch) name = decodeURIComponent(dnMatch[1]).replace(/\+/g, ' ');
    }
    if (!name) return;

    const text = $row.text();
    const sizeMatch = text.match(/([\d,.]+\s*(?:GB|MB|KB|TB|GiB|MiB))/i);
    const sizeText = sizeMatch ? sizeMatch[1] : '';

    let seeders = 0;
    const cols = $row.find('td');
    if (cols.length >= 3) seeders = parseSeeders($(cols[cols.length - 2]).text());

    results.push({
      name,
      size: sizeText || 'N/A',
      size_gb: parseSize(sizeText),
      seeders,
      magnet,
      provider: NAME,
    });
  });
  return results;
}

module.exports = { name: NAME, base_url: BASE, search };

File: scrapers/torrentquest.py
"""TorrentQuest scraper - HTML scraping."""

import re
from urllib.parse import quote
from scrapers.base import BaseScraper
from scrapers import client, utils


class Source(BaseScraper):
    name = 'TorrentQuest'
    base_url = 'https://torrentquest.com'

    def search(self, query):
        url = f'{self.base_url}/search?q={quote(query)}&sort=seeders&order=desc'
        soup = client.get_soup(url)
        if not soup:
            return []
        results = []
        for row in soup.select('table tbody tr, div.search-result'):
            magnet_tag = row.select_one('a[href^="magnet:"]')
            if not magnet_tag:
                continue
            magnet = magnet_tag['href']
            name_tag = row.select_one('a.title, td a[href*="/torrent/"]')
            name = utils.clean_name(name_tag.get_text()) if name_tag else ''
            if not name:
                dn_match = re.search(r'dn=([^&]+)', magnet)
                if dn_match:
                    from urllib.parse import unquote
                    name = unquote(dn_match.group(1)).replace('+', ' ')
            if not name:
                continue
            text = row.get_text()
            size_match = re.search(r'([\d,.]+\s*(?:GB|MB|KB|TB|GiB|MiB))', text, re.I)
            size_text = size_match.group(1) if size_match else ''
            seeders = 0
            cols = row.select('td')
            if len(cols) >= 3:
                seeders = utils.parse_seeders(cols[-2].get_text())
            results.append({
                'name': name,
                'size': size_text or 'N/A',
                'size_gb': utils.parse_size(size_text),
                'seeders': seeders,
                'magnet': magnet,
                'provider': self.name,
            })
        return results
